smtp/version.lua

```diff
@@ -0,0 +1,4 @@
+-- Сontains the module version.
+-- Requires manual update in case of release commit.
+
+return '0.0.6'
```

Comment:
We decided not to mark intermediate versions since it sounds rather bothersome to support. But it is recommended to make a new module release after this PR.

Reply:
The following command will install a version, which reports itself as 0.0.6, but it is far beyond 0.0.6:

```
tarantoolctl rocks install smtp
```

It looks not okay to report 0.0.6 for something that is not 0.0.6. If the version reporting may produce incorrect result (from a user perspective), the feature has no sense. (And if it has no sense, it doesn't matter how much effort it requires.)

I would use maven-release-plugin approach instead, see [here](https://github.com/tarantool/tarantool-java/commits/master).

![image](https://user-images.githubusercontent.com/92749/226407351-d0f914e9-345e-4019-9598-78f7f548622a.png)

IOW,

* Write `0.0.7` into version.lua, tag this commit, make a release.
* Write `0.0.8-dev`, or `0.0.8-snapshot` or `scm-1` into version.lua, push as another commit.
* Continue development.

Reply:
I engaged @LeonidVas into the discussion. Since we plan to make a release soon we agreed to proceed with current solution and return here back a bit later (discuss again). My concern is related to the scm-1 version is it seems acceptable to fix it later (because, unlike released versions, it is possible to fix later). I hope that not some many users will install the scm-1 version during this window.